infra/base-images/base-builder/detect_repo.py

```diff
@@ -55,7 +57,14 @@ def main():
   else:
     src_dir = os.environ.get('SRC', '/src')
 
-  for single_dir in os.listdir(src_dir):
+  dirs_to_search = os.listdir(src_dir)
+  if os.path.exists(GO_PATH) and args.repo_name:
+    for root, dirs, _ in os.walk(GO_PATH):
+      for test_dir in dirs:
+        if args.repo_name in test_dir:
+          dirs_to_search.append(os.path.join(root, test_dir))
+
+  for single_dir in dirs_to_search:
```

Comment:
I'd make a `search_dirs` function, put the rest of this function in it, and replace with a call to `search_dirs`

Reply:
Done.